### ISSUE TITLE
lt dataizes twice

### ISSUE THREAD

**Issue by levBagryansky** [objectionary/eo#1299]
@Graur please take a look at this case
```
[] > less-than
  assert-that > @
    seq
      QQ.io.stdout "SEQ STARTING\n"
      100
    $.less-than 300
```
This code works this way
```
[INFO] Running EOorg.EOeolang.EOthreads.EOless_thanTest
SEQ STARTING
SEQ STARTING
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.012 s - in EOorg.EOeolang.EOthreads.EOless_thanTest
```
I don't guess it must work this way.

**Comment by includealex** [comment]
@Graur @levBagryansky I think that it works correctly. Because when we use seq, in assert-that we compare everything what is inside seq scope to the given value. That is why this program
```
[] > app
  seq > x!
    stdout
      "你好\n"
    2
  or. > @
    (x.lt 1)
    (x.eq 2)
```
compares the whole scope of `seq`, dataizing `stdout` which is inside. An output will be
```bash
你好
你好
```

If user wants to compare object, which is returned from the `seq`, `@` should be used. For instance, `app` given above should be modified:
```
[] > app-modified
  seq > x!
    stdout
      "你好\n"
    2
  x.@ > y!
  or. > @
    (y.lt 1)
    (y.eq 2)
```
And in this case `stdout` will be called only once. And output will be:
```bash
你好
```

**Comment by levBagryansky** [comment]
@includealex Can you explain how to code this test:
```
[] > less-than
  assert-that > @
    seq
      QQ.io.stdout "SEQ STARTING\n"
      100
    $.less-than 300
```
?

**Comment by includealex** [comment]
@levBagryansky sure
```eo
[] > less-than
  seq > x
    QQ.io.stdout "SEQ STARTING\n"
    100
  assert-that > @
    x.@
    $.less-than 300

```

**Comment by levBagryansky** [comment]
@includealex I don't think it is ok that 
```
[] > less-than
  assert-that > @
    seq
      QQ.io.stdout "SEQ STARTING\n"
      100
    $.equal-to 100
```
dataizes once but 
```
[] > less-than
  assert-that > @
    seq
      QQ.io.stdout "SEQ STARTING\n"
      100
    $.less-than 300
```
May be we can edit `less-than` implementation to do this `x.@` stafff there?
Now `less-than`'s behaviour unexpecteble for users

**Comment by includealex** [comment]
@levBagryansky we can not edit `lt` implementation to do `x.@` staff. We have no ability to write `x.@ > y!` because attribute `ϕ` is absent.

**Comment by includealex** [comment]
@levBagryansky I think that using `stdout` in `seq` should lead to undefined number of calling `stdout` due to `seq` dataizes every time user refers to it.

**Comment by levBagryansky** [comment]
> @levBagryansky I think that using `stdout` in `seq` should lead to undefined number of calling `stdout` due to `seq` dataizes every time user refers to it.

There can be not just `stdout` but some important commands, that must be performed limited times.

**Comment by levBagryansky** [comment]
> @levBagryansky we can not edit `lt` implementation to do `x.@` staff. We have no ability to write `x.@ > y!` because attribute `ϕ` is absent.

@includealex Does not every object have this attribute?

**Comment by includealex** [comment]
> > @levBagryansky I think that using `stdout` in `seq` should lead to undefined number of calling `stdout` due to `seq` dataizes every time user refers to it.
> 
> There can be not just `stdout` but some important commands, that must be performed limited times.

@levBagryansky  I've tried to break it. But it works correctly everywhere due to `seq` implementation. You can take a look to these [tests](https://github.com/objectionary/eo/pull/1324/files) that are working correctly.

**Comment by Graur** [comment]
@levBagryansky @includealex I think if it doesn't work for `seq` but works for everything else then it looks like a bug

**Comment by 0pdd** [comment]
@levBagryansky the puzzle [#1370](https://github.com/objectionary/eo/issues/1370) is still not solved.

**Comment by 0pdd** [comment]
@levBagryansky the puzzle [#1953](https://github.com/objectionary/eo/issues/1953) is still not solved; solved: [#1370](https://github.com/objectionary/eo/issues/1370).

**Comment by 0pdd** [comment]
@levBagryansky all 2 puzzles are solved here: [#1370](https://github.com/objectionary/eo/issues/1370), [#1953](https://github.com/objectionary/eo/issues/1953).
